### PR TITLE
Added --es5 option to create a ES5 compatible compilations

### PR DESCRIPTION
**Pull request by maxbarnas**
Fixes: https://github.com/ckeditor/ckeditor5-dev-compiler/issues/18.

Note to self: Have to update Wiki.

### CODE REVIEW

**Comment by Reinmar** [comment]
LGTM. I only miss some tests. Compiler is super important for us so let's keep the CC high.

**Comment by Reinmar** [comment]
OK, there's a question now – how to include the regenerator in each test. For PoC we simply added it to Bender config, but this means that you'd need to change the configuration just to check ES5 version of the code. This is acceptable because we'll be mostly testing ES6 version as FF, Chrome, Edge and Safari (10, so the next version) has nearly 100% support and switch to ES5 only to confirm that after bundling everything should be fine or to test older browsers one day, but if you see a simple way to add regenerator in the compiler, then let's do this.

E.g. we could load it in https://github.com/ckeditor/ckeditor5-dev-compiler/blob/d43c2e6f12604ba7cd56730024da20bfc3ebbdde/lib/compiler/utils.js#L39?

File: lib/compiler/utils.js [hunk 1]
```diff
@@ -45,14 +45,26 @@ require( [ 'tests' ], bender.defer(), function( err ) {
 `,
 
 	/**
+	 * Adds polyfill to launcher code, when needed.
```
Comment (Reinmar) [inline lib/compiler/utils.js]: It doesn't add anything. It returns something ;)

Comment (maxbarnas) [inline lib/compiler/utils.js]: Figure of speech, man! ;)

**Comment by Reinmar** [comment]
All good. I only realised that in ckeditor5 we'll need to install babel-polyfill to make sure that it's accessible there... But I guess that we can ignore this issue, because when you'll install dev-compiler in a normal way, then this package should be hoisted to the main node_modules anyway.

**Comment by Reinmar** [comment]
Argh... I forgot to check whether you added babel-polyfil to this package's package.json :P. Will post-fix this in a separate commit.

**Comment by Reinmar** [comment]
OK, we decided to move this dependency to ckeditor5 anyway. I removed it in this repo and will add it in ckeditor5.

**Pull request merged**